stellargraph/layer/graphsage.py

```diff
@@ -808,14 +841,22 @@ def _get_sizes_from_generator(self, generator):
         Args:
              generator: The supplied generator.
         """
-        self.n_samples = generator.generator.num_samples
+        if not isinstance(generator, (GraphSAGELinkGenerator, GraphSAGENodeGenerator)):
+            raise TypeError(
+                "Generator should be an instance of GraphSAGELinkGenerator or GraphSAGENodeGenerator"
```

Comment:
swap the order to "GraphSAGENodeGenerator or GraphSAGELinkGenerator"?

Reply:
Seems pretty minor. Made the change.